app/src/main/java/com/example/hellorecycler/MyAdapter.kt

```diff
@@ -3,57 +3,61 @@ package com.example.hellorecycler
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import android.widget.ImageView
 import android.widget.TextView
 import androidx.recyclerview.widget.RecyclerView
+import kotlinx.android.synthetic.main.candy_text_layout.view.*
 
-// private val myDataset: Array<String>??
+// private val myDataset: Array<String>?? i get it now that the primary constructor creates the property
+// no additional declaration needed
 
-class MyAdapter(private val myDataset: List<String>) :
+class MyAdapter(private val myCandies: List<CandyData>, val clickListener: (CandyData)) :
     RecyclerView.Adapter<RecyclerView.ViewHolder>() { // rarely need type
 
-
     // provide a reference to the views for each data item
     // provide access to all the views for a data item in a view holder.
-    class MyViewHolder(val view: View) : RecyclerView.ViewHolder(view){
+    class MyViewHolder(val view: View) : RecyclerView.ViewHolder(view) {
         fun setText(label: String) {
             view.findViewById<TextView>(R.id.candy_text_view).apply {
                 text = label
             }
-
         }
-    }
-// this will be view group, result of inflation
 
-    // these 3 are required members of Adapter
+        // replace with:
+        fun bindData(candy: CandyData, clickListener: (CandyData)): Unit {
 
-    // Create new views (called by layout manager)
+            view.candy_text_view.text = candy.candyName
 
+            view.setOnClickListener { clickListener(candy) }
```

Comment:
getting "Expression 'clickListener' of type 'CandyData' cannot be invoked as a function. The function 'invoke()' is not found" here. Not sure if this is best place to set it.

Reply:
so this gets changed to candyClicked(candy)?